lib/python/qmk/makefile.py

```diff
@@ -0,0 +1,82 @@
+""" Functions for working with Makefiles
+"""
+import os
+
+import qmk.path
+from qmk.errors import NoSuchKeyboardError
+
+
+def parse_rules_mk_file(file, rules_mk=None):
+    """Turn a rules.mk file into a dictionary.
+
+    Args:
+        file: path to the rules.mk file
+        rules_mk: already parsed rules.mk the new file should be merged with
+
+    Returns:
+        a dictionary with the file's content
+    """
+    if not rules_mk:
+        rules_mk = {}
+
+    if os.path.exists(file):
+        rules_mk_lines = qmk.path.file_lines(file)
```

Comment:
If you use pathlib here you don't need a custom function inside qmk. Add a `from Pathlib import Path` and then throw a `file = Path(file)` at the top of this function. That way this function will work with string paths and pathlib objects.

```suggestion
        rules_mk_lines = file.read_text()
```